dynawo/sources/Util/DYNVisibility.h

```diff
@@ -0,0 +1,23 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+#ifndef UTIL_DYNVISIBILITY_H_
+#define UTIL_DYNVISIBILITY_H_
+
+#if defined _WIN32
+#define DLL_PUBLIC __declspec(dllexport)
```

Comment:
I though we wanted to avoid a solution with those kind of macros?

Reply:
When we use MODULE instead of SHARED, in add_library(dynawo_SolverIDA MODULE ...), cmake does not make the symbol of the factories visible so we are forced to do it for the dll API (4 functions in each solver or model)

Reply:
ok

Reply:
then why do we changed SHARED to MODULE?

Reply:
from cmake docs: MODULE libraries are plugins that are not linked into other targets but may be loaded dynamically at runtime using dlopen-like functionality.